Catappult/appcoins-sdk-ios
# 2819 query supported currencies

What does this PR do?

This PR adds a new service to obtain a list of supported currencies. Each element of this list will be represent by Currency new entity. It will also add a CurrencyUseCases and a CurrencyRepository.

Database changed?

No

Where should the reviewer start?

 Currency.swift
 CurrencyUseCases.swift
 CurrencyRepository.swift
 AppCoinsBillingClient.swift

How should this be manually tested?

Try to call the new request called getSupportedCurrencies() and verify if it is returning a list of Currency elements. Also you can make the request with this url ("\(endpoint)/currencies?type=FIAT&icon.height=128&limit=10") and verify if the method is returning the same list.

What are the relevant tickets?

Tickets related to this pull-request: APP-2819

Questions:

Does this add new dependencies which need to be added? (Eg. new keys, etc.)

Code Review Checklist

 Documentation on public interfaces
 Database changed? If yes - Migration?
 Remove comments & unused code & forgotten testing Logs
 Codestyle
 New Kotlin code has unit tests
 New flows in presenters unit tests
 Mappers/Validators with any kind of logic unit tests
 Functional tests pass
